openarm/netcan/transport.py

```
@@ -1,70 +1,99 @@
-        self.r = self.sock.makefile("r")
 
-    def encode(self, message: Message) -> None:
-        """Encode CAN message and send to stored writer."""
-        payload = {
-            "arbitration_id": message.arbitration_id,
-            "data": message.data.hex() if message.data else "",
-            "timestamp": message.timestamp if message.timestamp else time(),
-            "is_extended_id": message.is_extended_id,
-        }
-
-        data = json.dumps(payload) + "\n"
-        self.sock.sendall(data.encode("utf-8"))
-
-    def decode(self) -> Message | None:
-        """Decode and return a CAN message from stored reader."""
-        line = self.r.readline()
-        if not line:
+class AsyncSocketTransport:
+    """Async CAN transport for Socket using asyncio streams."""
+
+    def __init__(
+        self, reader: asyncio.StreamReader, writer: asyncio.StreamWriter
+    ) -> None:
+        """Initialize Async Socket Transport.
+
+        Args:
+            reader: Asyncio stream reader for incoming data
+            writer: Asyncio stream writer for outgoing data
+
+        """
+        self.reader = reader
+        self.writer = writer
+        self.addr = writer.get_extra_info("peername")
+
+    async def encode(self, message: Message) -> None:
+        """Encode CAN message and send to client.
+
+        Args:
+            message: CAN message to encode and send
+
+        """
+        try:
+            payload = {
+                "arbitration_id": message.arbitration_id,
+                "data": message.data.hex() if message.data else "",
+                "timestamp": message.timestamp if message.timestamp else time(),
+                "is_extended_id": message.is_extended_id,
+            }
+
+            data = json.dumps(payload) + "\n"
+            self.writer.write(data.encode("utf-8"))
+            await self.writer.drain()
+        except (ConnectionError, OSError):
+            # Connection closed, will be handled by caller
+            pass
```

The comment seems wrong, this will ignore the exception, caller have no way to know if messages that sent to client failed or not.